compiler/bindings/python/test/ir/tuner_test.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2025 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from iree.compiler import ir
+from iree.compiler.dialects import iree_codegen
+
+
+def run(fn):
+    with ir.Context(), ir.Location.unknown():
+        module = ir.Module.create()
+        with ir.InsertionPoint(module.body):
+            print("\nTEST:", fn.__name__)
+            fn()
+    return fn
+
+
+@run
+def root_op():
+    module_str = """
+        module {
+            func.func @matmul(%arg0: tensor<4x4xf32>, %arg1: tensor<4x4xf32>) -> tensor<4x4xf32> {
+                %cst = arith.constant 0.000000e+00 : f32
+                %0 = tensor.empty() : tensor<4x4xf32>
+                %1 = linalg.fill { root_op } ins(%cst : f32) outs(%0 : tensor<4x4xf32>) -> tensor<4x4xf32>
+                %2 = linalg.matmul { root_op } ins(%arg0, %arg1 : tensor<4x4xf32>, tensor<4x4xf32>) outs(%1 : tensor<4x4xf32>) -> tensor<4x4xf32>
+                return %2 : tensor<4x4xf32>
+            }
+        }
+    """
+    input_module = ir.Module.parse(module_str)
+    root_op_list = iree_codegen.get_tuner_root_ops(input_module)
+    assert len(root_op_list) == 2
+    assert root_op_list[0].name == "linalg.fill"
+    assert root_op_list[1].name == "linalg.matmul"
+
+    del root_op_list[0].attributes["root_op"]
```

Comment:
I don't think python is supposed to modify IR like this [1]: there may be some dangling references to IR objects. Instead, can you make it a few separate test cases with their own individual inputs?

[1] https://mlir.llvm.org/docs/Bindings/Python/#ownership-in-the-core-ir